### ISSUE TITLE
Enable Deployment Prerequisites fails because Set-AzureRmVMCustomScriptExtension returns object with different properties than expected

### ISSUE THREAD

**Issue by michalkovy** [Microsoft/vsts-tasks#1788]
My deployment to azure fails with logs:

2016-05-24T15:54:07.4176178Z [Azure Call]Getting the custom script extension 'WinRMCustomScriptExtension' for vm 'MyWindowsVM'
2016-05-24T15:54:11.2670027Z ##[debug]IsExtensionExists: False
2016-05-24T15:54:11.2740034Z [Azure Call]Setting the custom script extension 'WinRMCustomScriptExtension' for vm 'MyWindowsVM'
2016-05-24T15:55:19.3738127Z [Azure Call]Set the custom script extension 'WinRMCustomScriptExtension' for vm 'MyWindowsVM'
2016-05-24T15:55:19.3928146Z [Azure Call]Removing the custom script extension 'WinRMCustomScriptExtension' for vm 'MyWindowsVM'
2016-05-24T15:56:24.2553002Z [Azure Call]Removed the custom script extension 'WinRMCustomScriptExtension' for vm 'MyWindowsVM'
2016-05-24T15:56:24.2763023Z ##[error]Failed to enable deployment prerequisites. Unable to set the custom script extension 'WinRMCustomScriptExtension' for virtual machine 'MyWindowsVM': 

I was trying to run the script manually on my machine and found the following command has a different properties then the script expects, this:
$result = Set-AzureRmVMCustomScriptExtension -resourceGroupName $resourceGroupName -vmName $vmName -name $extensionName -fileUri $configWinRMScriptFile, $makeCertFile, $winrmConfFile  -run $scriptToRun -argument $dnsName -location $location

returns me this:

RequestId IsSuccessStatusCode StatusCode ReasonPhrase
--------- ------------------- ---------- ------------
                         True         OK OK          

But the Utility.ps1 script has this check after the command:
if($result.Status -ne "Succeeded")
{

As you can see, such condition in if will return true because there isn't a property Status. I have e.g. a Boolean IsSuccessStatusCode there instead.

**Comment by codito** [comment]
@michalkovy what is the version of azure powershell on the agent machine?

**Comment by michalkovy** [comment]
I see Azure and AzureRM modules to be of version 1.4.0. I installed it last week.

**Comment by codito** [comment]
Can you please try with `1.3.2` azure powershell?

It's on our backlog to add support for `1.4.0`.

**Comment by michalkovy** [comment]
The 1.3.2 versions didn't work either, I tried 1.3.0 and 1.2.2 and had the same bad result, too. Then I tried 0.9.8 and it started to work.

Supporting 1.4.0 won't fix the more general issue that you have and it is that basically your product depends on a specific version of modules and there isn't any version enforcement in place.  Ideally I would like if it got the version it needs automatically.

The other related problem you have is that the tasks from on premises version of TFS  have a link "More Information" that links to generic github page about the module and there is a recommended version 1.3.0. I would expect that different version of TFS has a different recommended version of powershell.
